src/Shared/CopyOnWriteDictionary.cs

```diff
@@ -80,54 +65,48 @@ internal CopyOnWriteDictionary(IEqualityComparer<K> keyComparer)
         /// <summary>
         /// Constructor taking a specified comparer for the keys and an initial capacity
         /// </summary>
-        internal CopyOnWriteDictionary(int capacity, IEqualityComparer<K> keyComparer)
+        internal CopyOnWriteDictionary(int capacity, IEqualityComparer<K>? keyComparer)
```

Comment:
Just removing the `?` wouldn't compile because this constructor is explicitly passed `null` a few lines above and `ImmutableDictionary.Create` actually does accept `null`. I'm not sure if nullable reference types, as introduced in C# 8.0, have already made it into official docs and all APIs are annotated accordingly. Judging by this occurrence it is probably not the case.